src/vmspawn/vmspawn.c

```diff
@@ -0,0 +1,475 @@
+               "     --load-credential=ID:PATH\n"
+               "                            Load credential to pass to container from file or\n"
+               "                            AF_UNIX stream socket.\n"
+               "\nSee the %2$s for details.\n",
+               program_invocation_short_name,
+               link,
+               ansi_underline(),
+               ansi_normal(),
+               ansi_highlight(),
+               ansi_normal());
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        enum {
+                ARG_VERSION = 0x100,
+                ARG_NO_PAGER,
+                ARG_QEMU_SMP,
+                ARG_QEMU_MEM,
+                ARG_QEMU_KVM,
+                ARG_QEMU_GUI,
+                ARG_SET_CREDENTIAL,
+                ARG_LOAD_CREDENTIAL,
+        };
+
+        static const struct option options[] = {
+                { "help",            no_argument,       NULL, 'h'                 },
+                { "version",         no_argument,       NULL, ARG_VERSION         },
+                { "no-pager",        no_argument,       NULL, ARG_NO_PAGER        },
+                { "image",           required_argument, NULL, 'i'                 },
+                { "qemu-smp",        optional_argument, NULL, ARG_QEMU_SMP        },
+                { "qemu-mem",        optional_argument, NULL, ARG_QEMU_MEM        },
+                { "qemu-kvm",        optional_argument, NULL, ARG_QEMU_KVM        },
+                { "qemu-gui",        no_argument,       NULL, ARG_QEMU_GUI        },
+                { "set-credential",  required_argument, NULL, ARG_SET_CREDENTIAL  },
+                { "load-credential", required_argument, NULL, ARG_LOAD_CREDENTIAL },
+                {}
+        };
+
+        int c, r;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        optind = 0;
+        while ((c = getopt_long(argc, argv, "+hi:", options, NULL)) >= 0)
+                switch (c) {
+                case 'h':
+                        return help();
+
+                case ARG_VERSION:
+                        return version();
+
+                case 'i':
+                        r = parse_path_argument(optarg, false, &arg_image);
+                        if (r < 0)
+                                return r;
+
+                        arg_settings_mask |= SETTING_DIRECTORY;
+                        break;
+
+                case ARG_NO_PAGER:
+                        arg_pager_flags |= PAGER_DISABLE;
+                        break;
+
+                case ARG_QEMU_SMP:
+                        arg_qemu_smp = strdup(optarg);
+                        if (!arg_qemu_smp)
+                                log_oom();
+                        break;
+
+                case ARG_QEMU_MEM:
+                        arg_qemu_mem = strdup(optarg);
+                        if (!arg_qemu_mem)
+                                log_oom();
```

Comment:
should be fatal, i.e. `return log_oom()`, above the same

Reply:
Fixed in 7e39113551.